client/src/components/NavHeader.tsx

```diff
@@ -1,89 +1,79 @@
-import React, { useState, useEffect } from "react";
+import React from "react";
 import {
-  Button,
-  ButtonGroup,
-  Dropdown,
-  Container,
-  Navbar,
-  Nav,
+	Button,
+	ButtonGroup,
+	Dropdown,
+	Container,
+	Navbar,
+	Nav,
 } from "react-bootstrap";
+import { ethers } from "ethers";
 import { LinkContainer } from "react-router-bootstrap";
+import en from "../translations/en.json";
 
 interface IProps {
-  connectedWallet?: any; 
-  provider?: any; 
-  onWalletConnectClick?: any; 
-  onDisconnectWallet?: any; 
+  connectedWallet?: string;
+  provider?: ethers.providers.Web3Provider;
+  onWalletConnectClick?: any;
+  onDisconnectWalletClick?: any;
 }
 
-interface IState {
-  creatorAddress?: string; 
-  amountValue?: number;
-  sentTransactionHash?: string;
-  transactionModalState?: number; 
-  aaveRate?: string; 
-  contract?: any; 
-}
-
-const NavHeader = ({connectedWallet, provider, onWalletConnectClick, onDisconnectWallet}: IProps) => {
-  
-  const [creatorAddress, setCreatorAddress] = useState(""); 
-  const [amountValue, setAmountValue] = useState(0); 
+const NavHeader = ({
+	connectedWallet, onWalletConnectClick, onDisconnectWalletClick,
```

Comment:
Do we no longer need the `provider` in the constructor here? If not, we can also remove it from `IProps`